torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -65,6 +70,12 @@ void IRPrinter::printHeader(Fusion* fusion, const std::string& kernel_name_) {
     indent();
     os << "Philox rnd(seed, idx, offset);\n";
   }
+  if (fusion->hasBlockReduction() || fusion->hasGridReduction()) {
+    indent();
+    // TODO: Dynamic sizing possible? blockReduce originally used 1024
+    // values of a given type
+    os << "__shared__ float shared_mem[1024];\n";
+  }
```

Comment:
Here, I changed the allocation of shared memory to be done in the entry global function so that one single allocation can be used in both `blockReduce` and `globalReduce`.